### ISSUE TITLE
Weird highlight boxes

### ISSUE THREAD

**Issue by giuli635** [koreader/koreader#11644]
* KOReader version: 2024.03.1
* Device: Android 11 and Arch Linux

#### Issue

Weird highlight boxes created. This happens throughout the document, but more frequently when is near a math expression.

![image](https://github.com/koreader/koreader/assets/75577897/402f3025-f054-40c9-8703-9cd40e36eced)

#### Steps to reproduce

I just don't know if it's legal uploading the pdf jajajaja.

**Comment by Frenzie** [comment]
Presumably that's just the math saying it's that tall, unless you're saying it doesn't happen in other viewers.

**Comment by giuli635** [comment]
It doesn't happen in Evince for example. I just checked if its a rendering issue but it also saves the highlight that way. But as I said, in Evince I dont have this problem.

**Comment by Frenzie** [comment]
Based on some random PDF (<https://arxiv.org/pdf/2305.17493.pdf>) I'd say it's a bit of a toss up what happens to come out better where. It does generally seem to come out a bit nicer in Firefox, probably mostly as a happy coincidence due to how they lay out text. (It's regular HTML with fancy positioning, which is fundamentally quite different from how native PDF implementations work.)

The (hopefully) good news is that there are likely to be some improvements here in the future regardless, due to various updated dependencies.

## Okular
<img src=https://github.com/koreader/koreader/assets/202757/7293592c-65d1-4866-9ced-c7572a01647e>

<img src=https://github.com/koreader/koreader/assets/202757/e305a189-4c8d-4858-af97-b667dd41e736 width=45%>

## Evince, KOReader on top
<img src=https://github.com/koreader/koreader/assets/202757/fa1e0a65-10f1-484a-b588-17ba659806d7 width=45%>

**Comment by Frenzie** [comment]
> The (hopefully) good news is that there are likely to be some improvements here in the future regardless, due to various updated dependencies.

You can probably get a preview of that in a newer MuPDF app (whether on desktop or Android), unless it doesn't expose text selection.

**Comment by giuli635** [comment]
Okay, I'm doing a last recap with all the info I realize could be useful, if you tell me I just have to wait, I'll wait :)

#### Evince selection before any modification in KOReader

![image](https://github.com/koreader/koreader/assets/75577897/a289cb32-162f-4945-ba8c-ef15bc879577)

#### Evince opening the file modified by KOReader

![image](https://github.com/koreader/koreader/assets/75577897/4bbcf918-15f4-41a6-958e-f0d210b794d7)

#### KOReader selection before any modification

![image](https://github.com/koreader/koreader/assets/75577897/1c5a4bad-5599-4286-a6d4-b85975981b28)

Doing this, when I want to use reflow and then select in desktop it crashes, but I don't have time to see the logs, I'll see what's going on later.

#### The page of the pdf

I leave here the page of the pdf [test.pdf](https://github.com/koreader/koreader/files/14922791/test.pdf).

**Comment by Frenzie** [comment]
Curious, the geometry doesn't seem out of the ordinary in the PDF but as shown the selection returned is rather tall.

(Some relevant output from `mutool trace`.)
```
    <span font="DLYZCT+LatinModernMath-Regular" wmode="0" bidi="0" trm="9.96264 0 0 9.96264">
        <g unicode="𝜆" glyph="4470" x="236.69" y="353.349" adv=".583"/>
        <g unicode="𝑥" glyph="1319" x="242.49822" y="353.349" adv=".572"/>
        <g unicode="." glyph="15" x="248.19684" y="353.349" adv=".278"/>
        <g unicode="𝑥" glyph="1319" x="250.96645" y="353.349" adv=".572"/>
        <g unicode="𝑧" glyph="1321" x="256.66508" y="353.349" adv=".465"/>
    </span>
    <span font="IWOTBY+LibreBaskerville-Regular" wmode="0" bidi="0" trm="10.161893 0 0 9.96264">
        <g unicode="a" glyph="66" x="264.867" y="353.349" adv=".554"/>
        <g unicode="n" glyph="79" x="270.4967" y="353.349" adv=".689"/>
        <g unicode="d" glyph="69" x="277.49827" y="353.349" adv=".675"/>
    </span>
```

```
04/09/24-21:40:58 DEBUG dict lookup word: 𝜆𝑥𝑦.𝑦𝑥 {
  "124x270+497+545"
} --[[table: 0x7c9d79e20440]]
```

But yeah, at the moment there's a good chance it'll be fixed automatically once MuPDF is finally updated.

**Comment by benoit-pierre** [comment]
Nope, I get the same tall selection boxes with a build using MuPDF 1.24.1.

**Comment by giuli635** [comment]
So is a problem of MuPDF? Not of KOReader?

**Comment by Frenzie** [comment]
Yes, here's a screenshot from MyPDF 1.24.
![Screenshot_2024-04-09-22-51-51-80_c27c9384626a2c2f1677dcc28c11a148](https://github.com/koreader/koreader/assets/202757/f5470f69-e425-41af-9a3a-cd8d35f8f701)

**Comment by Frenzie** [comment]
As a random thought, I think the Latin Modern Math font had a tall lineheight issue about a decade ago. Although that wouldn't explain why only MuPDF has this issue.

**Comment by giuli635** [comment]
So, maybe I should create an issue in MuPDF?

**Comment by Frenzie** [comment]
I suspect this issue already covers it: https://bugs.ghostscript.com/show_bug.cgi?id=700466

Note how the screenshot of http://beta.hebrewbooks.org/pagefeed/hebrewbooks_org_9717_1.pdf shows a very tall selection in the header, when the box actually hugs the Hebrew characters.

<img src=https://github.com/koreader/koreader/assets/202757/2cf6dd95-ff85-4b93-86e9-e8e90f581855 width=30%>

**Comment by giuli635** [comment]
That's five years old, do you think this will be addressed?

**Comment by Frenzie** [comment]
Your test case could be useful because it shows a clear contrast between desired and undesired behavior. I think it'd be worth adding there, unless @benoit-pierre has different thoughts.

**Comment by benoit-pierre** [comment]
I would report the issue upstream, yes.

**Comment by giuli635** [comment]
Thank you guys, you've been very helpful! Do I leave the issue open?

**Comment by Frenzie** [comment]
Let's leave it open for now.

**Comment by Chlorpheniramine** [comment]
> * KOReader version: 2024.03.1
> * Device: Android 11 and Arch Linux
> 
> #### Issue
> Weird highlight boxes created. This happens throughout the document, but more frequently when is near a math expression.
> 
> ![image](https://private-user-images.githubusercontent.com/75577897/320678724-402f3025-f054-40c9-8703-9cd40e36eced.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQxMzA0MDAsIm5iZiI6MTcxNDEzMDEwMCwicGF0aCI6Ii83NTU3Nzg5Ny8zMjA2Nzg3MjQtNDAyZjMwMjUtZjA1NC00MGM5LTg3MDMtOWNkNDBlMzZlY2VkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI2VDExMTUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiYzRjZTBiMWUyNmE0ZTAwOGQ4ODJkMDkyOWY2ZDE1YTcwY2ViNTJmNmE2OWM2ZmM0N2U4ODZiOWY1MGQzY2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6AT6GfhTrFUNyP2JOkahzKFwzsRa-Z5mWC0WD1VN0Vw)
> 
> #### Steps to reproduce
> I just don't know if it's legal uploading the pdf jajajaja.

#11716
This look quite similar to the condition I encountered.  I propose to merge it to this thread and close it if necessary.

**Comment by benoit-pierre** [comment]
It looks like this got fixed in 1.25.4:

> Improve font Ascent/Descent handling.

With possible further improvements in 1.26.0:

> Add some more sanity checks for font ascent/descent values
